ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -264,28 +255,36 @@ Result conv2d(
         return {conv_output, output_height, output_width, weight_tensor_on_device, bias_tensor_on_device};
     } else {
         // run conv as matmul
-        std::optional<ttnn::operations::matmul::MatmulProgramConfig> program_config = std::nullopt;
-        std::optional<MemoryConfig> mm_output_memory_config = std::nullopt;
-        if (input_tensor_post_tm.is_sharded()) {
-            uint32_t num_cores_c = get_num_cores_channels_from_parallel_config(parallel_config);
-            program_config = determine_matmul_op_config_from_conv_op_config(
-                opt_conv_op_parallel_config,
-                opt_conv_op_block_config,
-                parallel_config.shard_scheme == TensorMemoryLayout::HEIGHT_SHARDED,
-                conv_config.activation,
-                parallel_config.shard_orientation == ShardOrientation::COL_MAJOR,
-                num_cores_c);
-            mm_output_memory_config = conv_out_memory_config;
```

Comment:
Pls dont' revert conv2d related stuff, 
we should decouple conv changes and matmul changes